sandbox/cgroup/scanning.go

```diff
@@ -0,0 +1,152 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/snap/naming"
+)
+
+var (
+	// string that looks like a hook security tag
+	roughHookTagPattern = regexp.MustCompile(`snap\.[^.]+\.hook\.[^.]+`)
+	// string that looks like an app security tag
+	roughAppTagPattern = regexp.MustCompile(`snap\.[^.]+\.[^.]+`)
+)
+
+// securityTagFromCgroupPath returns a security tag from cgroup path.
+func securityTagFromCgroupPath(path string) (securityTag string) {
+	leaf := filepath.Base(filepath.Clean(path))
+
+	// We are only interested in cgroup directory names that correspond to
+	// services and scopes, as they contain processes that have been invoked
+	// from a snap.
```

Comment:
Probbaly worth adding a comment that we have:
```
// Expected format of leaf name:
//   snap.<snap>.<svc>.service - assigned by systemd for services
//   snap.<snap>.<app>.<uuid>.scope - transient scope for apps
//   snap.<snap>.hook.<app>.<uuid>.scope - transient scope for hooks
```

Reply:
Done